src/PrestaShopBundle/Form/Admin/Improve/Shipping/Carrier/SizeWeightSettings.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Form\Admin\Improve\Shipping\Carrier;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\PositiveOrZero;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use PrestaShopBundle\Translation\TranslatorInterface;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Type;
+
+class SizeWeightSettings extends TranslatorAwareType
+{
+    /**
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param string $dimensionUnit
+     * @param string $weightUnit
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        private string $dimensionUnit,
+        private string $weightUnit,
+    ) {
+        parent::__construct($translator, $locales);
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        parent::buildForm($builder, $options);
+        $builder
+            ->add('max_width', IntegerType::class, [
+                'label' => $this->trans('Max order width', 'Admin.Shipping.Feature'),
+                'unit' => $this->dimensionUnit,
+                'label_help_box' => $this->trans('Maximum width managed by this carrier. Set the value to "0", or leave this field blank to ignore.', 'Admin.Shipping.Help') . ' ' . $this->trans('The value must be an integer.', 'Admin.Shipping.Help'),
```

Comment:
Maybe we should adapt the wording since the blank value is not allowed anymore (which makes sense actually)
```suggestion
                'label_help_box' => $this->trans('Maximum width managed by this carrier. Set the value to "0" to ignore.', 'Admin.Shipping.Help') . ' ' . $this->trans('The value must be an integer.', 'Admin.Shipping.Help'),
```

Reply:
Same for the other fields wording